tests/integration/seed/seedsVerification.test.js

```diff
@@ -147,18 +175,78 @@ describe('SEEDS VERIFICATION', () => {
           attendanceSheetList = await AttendanceSheet
             .find()
             .populate({ path: 'trainee', select: 'id', populate: { path: 'userCompanyList' } })
+            .populate({
+              path: 'course',
+              select: '_id type companies',
+              populate: { path: 'slots', select: 'startDate' },
+            })
+            .populate({ path: 'company', select: '_id' })
             .setOptions({ allCompanies: true })
             .lean();
         });
 
+        it('should pass if every course exists', () => {
+          const someCoursesDontExist = attendanceSheetList.some(a => !a.course);
+
+          expect(someCoursesDontExist).toBeFalsy();
+        });
+
+        it('should pass if only intra courses have date in attendance sheet', () => {
+          const everyIntraAttendanceSheetHasDate = attendanceSheetList
+            .every(a => a.course.type === INTER_B2B || a.date);
+
+          expect(everyIntraAttendanceSheetHasDate).toBeTruthy();
```

Comment:
Que penses-tu de plutot filtrer sur les INTRA et verifier que le champs date existe ?

Reply:
ça fait itérer 2 fois sur le tableau, pourquoi tu préfères faire comme ça ?

Reply:
Je trouve qu'ici la condition est un peu moins clair + si un jour on a pas que des inter et inter b2b il faudra modifier la condition. Mais c'est vrai que niveau perf c'est moins bien. Je ne sais pas ce qui est le mieux